src/Dependencies.jl

```diff
@@ -72,10 +72,14 @@ struct Dependency <: AbstractDependency
         if length(compat) > 0
             spec = Pkg.Types.semver_spec(compat) # verify compat is valid
             if build_version !== nothing && !(build_version in spec)
-                error("build_version and compat for $(pkg) are incompatible")
+                throw(ArgumentError("build_version and compat for $(pkg) are incompatible"))
             end
             if pkg.version != Pkg.Types.VersionSpec("*") && !(pkg.version in spec)
-                error("PackageSpec version and compat for $(pkg) are incompatible")
+                throw(ArgumentError("PackageSpec version and compat for $(pkg) are incompatible"))
+            end
+        else
+            if pkg.version != Pkg.Types.VersionSpec("*")
+                throw(ArgumentError("Use the `compat` kwarg to `Dependency` rather than the `version` kwarg to `PackageSpec`"))
```

Comment:
I think this will break BB; see the source code comments in my PR therr

Reply:
I added this to satisfy [this test](https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/123/files#diff-7191d1cf5dee1b5b6cdf1dbe4abba72c5880caa2f3f4a8f3d5c0726b07a3259bR50). Should the test be removed then?

Reply:
Sorry, I left that in by mistake. Should be fixed now.

(And I'd *like* to have that error, but that would require bigger changes in BB and BBB -- the split into two packages doesn't make this easier :/. Perhaps it can be addressed in a future PR, but for now I'd prefer to keep it simple, to reduce the risk of me introducing regressions.)